### DISCUSSION TITLE
Consider replace CMake with CMake Highlight to provide basic CMake language support

### DISCUSSION THREAD

**Discussion by ivaquero** [microsoft/vscode-cmake-tools#3553] [Ideas]
[CMake Highlight](https://github.com/zchrissirhcz/vscode-cmake-highlight) is a lightweight extension to provide CMake Highlighting and Completion, and it's under active development, I think it's good enough to replace the outdated bundled CMake extension.

I saw a closed CMake extension in the suggestion. I tried it, but it needs extra installation of .NET Runtime, a lightweight one is preferable.

**Comment by H-G-Hristov** [comment]
I think what you actually want are these:
https://github.com/microsoft/vscode-cmake-tools/issues/93
https://github.com/microsoft/vscode-cmake-tools/issues/534#issuecomment-1921478670
 You can vote for it an make some noise :)

**Comment by gcampbell-msft** [comment]
@ivaquero @H-G-Hristov @zchrissirhcz Thanks for the comments! We are actively working on supporting CMake language support (colorization, hover info, etc) to our own extension. We hope that in the next couple of releases we will have this support!

We have this support in Visual Studio and we plan to bring this to CMake Tools. Thanks!

**Reply by ivaquero** [reply]
Exciting news! Nice to hear from you! Looking forward to trying it soon.

**Comment by Zingam** [comment]
> @ivaquero @H-G-Hristov @zchrissirhcz Thanks for the comments! We are actively working on supporting CMake language support (colorization, hover info, etc) to our own extension. We hope that in the next couple of releases we will have this support!
> 
> We have this support in Visual Studio and we plan to bring this to CMake Tools. Thanks!

Will there be a configurable formatter for CMake?

**Comment by gcampbell-msft** [comment]
@ivaquero Thanks for this idea, we now support basic CMake language support in our extension! It is currently available in the pre-release version of the extension and will be available in the next official release once 1.20 officially releases, we hope this will be within the next month. 

Thanks!
